packages/styled-components/src/models/ThemeProvider.tsx

```diff
@@ -66,8 +66,12 @@ function mergeTheme(theme: ThemeArgument, outerTheme?: DefaultTheme): DefaultThe
   return outerTheme ? { ...outerTheme, ...theme } : theme;
 }
 
-export function useTheme(): DefaultTheme | undefined {
-  return useContext(ThemeContext);
+export function useTheme(): DefaultTheme {
+  const theme = useContext(ThemeContext);
+  if(!theme) {
+    throw Error('can not use useTheme outside a ThemeProvider');
```

Comment:
We have a `StyledError` class that should be used here for consistency. Can you also make sure you're applying prettier formatting please?

Reply:
I will update the PR with the suggested changes